lib/src/bridge/characteristics_mixin.dart

```diff
@@ -230,39 +252,47 @@ mixin CharacteristicsMixin on FlutterBLE {
     );
   }
 
-  Stream<CharacteristicWithValue> _createMonitoringStream(
+  Stream<CharacteristicWithValueAndTransactionId> _createMonitoringStream(
     void Function() onListen,
     bool Function(CharacteristicWithValueAndTransactionId) filter,
     Peripheral peripheral,
     String transactionId,
   ) {
-    Stream<CharacteristicWithValue> stream = _characteristicsMonitoringEvents
-        .map(
-          (rawJsonValue) =>
-              _parseCharacteristicWithValueWithTransactionIdResponse(
-                  peripheral, rawJsonValue),
-        )
+    Stream<CharacteristicWithValueAndTransactionId> stream =
+      _characteristicsMonitoringEvents
+        .map((rawValue) {
+          String rawJsonValue = "";
+          if (rawValue is String) {
+            rawJsonValue = rawValue;
+          }
```

Comment:
This can actually be get rid of by changing to `.invokeMethod<String>(...)`
And yes, it's guaranteed to be string at this point in time.